components/datasets/form/DatasetsForm.js

```diff
@@ -39,68 +60,64 @@ class DatasetsForm extends React.Component {
     });
   }
 
-  componentDidMount() {
-    const { dataset } = this.props;
-    // Get the dataset and fill the
-    // state with its params if it exists
-    if (dataset) {
-      fetchDataset(dataset, { includes: 'layer' })
-        .then((data) => {
-          const { provider, type, tableName, applicationConfig, layer } = data || {};
-          const appConfigRW = applicationConfig && applicationConfig[process.env.APPLICATIONS];
-          const publishedLayersOrder = appConfigRW && appConfigRW.publishedLayersOrder;
-
-          let layers = layer.filter(l => l.published);
-
-          // We are doing this check since most metadata objects might not have yet
-          // this field initialized
-          if (publishedLayersOrder && publishedLayersOrder.length > 0) {
-            layers = publishedLayersOrder.map(l => layers.find(e => e.id === l));
+  componentWillMount() {
+    const { dataset: datasetId } = this.props;
+    // Get the dataset and fill the state with its params if it exists
+    if (datasetId) {
+      fetchDataset(datasetId, {
+        includes: 'layer',
+        app: process.env.APPLICATIONS
+      })
+        .then((dataset) => {
+          const { provider, applicationConfig, layer: layers } = dataset;
+          let _layers = layers;
+
+          // sorts layers if applies
+          if (
+            applicationConfig &&
+            applicationConfig[process.env.APPLICATIONS] &&
+            applicationConfig[process.env.APPLICATIONS].layerOrder &&
+            layers.length > 1) {
+            const { layerOrder } = applicationConfig[process.env.APPLICATIONS];
+            _layers = sortLayers(layers, layerOrder);
           }
 
           this.setState({
-            form: this.setFormFromParams(data),
-            // Stop the loading
+            form: this.setFormFromParams(dataset),
             loading: false,
             loadingColumns: true,
-            publishedLayers: layers
+            layers: _layers,
+            dataDataset: dataset
           });
 
           if (provider !== 'wms') {
             // fetchs column fields based on dataset type
-            fetchFields({
-              id: dataset,
-              type,
-              provider,
-              tableName
-            })
-              .then((columns) => {
+            const url = getFieldUrl(dataset);
+            fetchFields(url)
```

Comment:
Removed the `fetchFields` function in services/dataset. Using one in `services/fields` instead.